pyrometheus/codegen/python.py

```diff
@@ -321,6 +327,33 @@ def get_mixture_internal_energy_mass(self, temperature, mass_fractions):
         emix = self.get_mass_average_property(mass_fractions, e0_rt)
         return self.gas_constant * temperature * emix
 
+    def get_mixture_entropy_mass(self, pressure, temperature, mass_fractions):
+        mmw = self.get_mix_molecular_weight(mass_fractions)
+        return 1.0/mmw * self.get_mixture_entropy_mole(pressure, temperature, <%
+                                                       %>mass_fractions)
+
+    def get_mole_average_property(self, mass_fractions, spec_property):
+        mmw = self.get_mix_molecular_weight(mass_fractions)
+        mole_fracs = self.get_mole_fractions(mmw, mass_fractions)
+        return sum([mole_fracs[i] * spec_property[i]
+                    for i in range(self.num_species)])
+
+    def get_mixture_enthalpy_mole(self, temperature, mass_fractions):
+        h0_rt = self.get_species_enthalpies_rt(temperature)
+        hmix = self.get_mole_average_property(mass_fractions, h0_rt)
+        return self.gas_constant * temperature * hmix
+
+    def get_mixture_entropy_mole(self, pressure, temperature, mass_fractions):
+        mmw = self.get_mix_molecular_weight(mass_fractions)
+        # necessary to avoid nans in the log function below
+        x = self.usr_np.where(
+            self.usr_np.less(self.get_mole_fractions(mmw, mass_fractions), 1e-16),
+            1e-16, self.get_mole_fractions(mmw, mass_fractions))
+        s0_r = self.get_species_entropies_r(pressure, temperature)
+        smix = self.get_mole_average_property(mass_fractions, s0_r)
+        xmix = self.get_mole_average_property(mass_fractions, self.usr_np.log(x))
+        return self.gas_constant * (smix - xmix)
```

Comment:
I don't like this notation. `xmix` does not right away indicate an entropy. We should have something like `s_t_mix` and `s_x_mix` to denote the individual contributions to the entropy.

Reply:
:+1: